src/Compilers/CSharp/Portable/BoundTree/Expression.cs

```diff
@@ -355,7 +265,8 @@ internal partial class BoundIndexerAccess : IIndexedPropertyReferenceExpression
 
         ISymbol IMemberReferenceExpression.Member => this.Indexer;
 
-        ImmutableArray<IArgument> IHasArgumentsExpression.ArgumentsInEvaluationOrder => BoundCall.DeriveArguments(this.Arguments, this.ArgumentNamesOpt, this.ArgsToParamsOpt, this.ArgumentRefKindsOpt, this.Indexer.Parameters, this.Syntax);
+        ImmutableArray<IArgument> IHasArgumentsExpression.ArgumentsInEvaluationOrder 
+            => BoundCall.DeriveArguments(this, this.BinderOpt, this.Indexer, this.Indexer.GetOwnOrInheritedGetMethod(),this.Arguments, this.ArgumentNamesOpt, this.ArgsToParamsOpt, this.ArgumentRefKindsOpt, this.Indexer.Parameters, this.Expanded, this.Syntax);
```

Comment:
>this.Indexer.GetOwnOrInheritedGetMethod() [](start = 77, length = 41)

This doesn't look right. What if this is a target of an assignment and the indexer is write-only?